rwengine/src/audio/SoundManager.cpp

```diff
@@ -108,9 +113,36 @@ void SoundManager::SoundSource::loadFromFile(const std::string& filename) {
         RW_ERROR("Couldn't open the audio codec context");
         return;
     }
+#else
+    // Initialize codec context for the decoder.
+    AVCodecContext* codecContext = avcodec_alloc_context3(codec);
+    if (!codecContext) {
+        av_frame_free(&frame);
+        avformat_close_input(&formatContext);
+        RW_ERROR("Couldn't allocate a decoding context.");
+        return;
+    }
+
+    // Fill the codecCtx with the parameters of the codec used in the read file.
+    if (avcodec_parameters_to_context(codecContext, audioStream->codecpar) != 0) {
+        avcodec_close(codecContext);
+        avcodec_free_context(&codecContext);
+        avformat_close_input(&formatContext);
+        RW_ERROR("Couldn't find parametrs for context");
+    }
+
+    // Initialize the decoder.
+    if (avcodec_open2(codecContext, codec, nullptr) != 0) {
+        avcodec_close(codecContext);
+        avcodec_free_context(&codecContext);
+        avformat_close_input(&formatContext);
+        RW_ERROR("Couldn't open the audio codec context");
+        return;
+    }
+#endif
 
     // Expose audio metadata
-    channels = codecContext->channels;
+    channels = kNrOfOutputChannels;
     sampleRate = codecContext->sample_rate;
```

Comment:
`WaveHeader` seems to use `uint32_t`, should this reflect it and cast or just cast?